test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -2607,6 +2607,47 @@ BOOST_AUTO_TEST_CASE(event_no_arguments)
 	BOOST_CHECK_EQUAL(m_logs[0].topics[0], dev::keccak256(string("Deposit()")));
 }
 
+BOOST_AUTO_TEST_CASE(events_with_same_name)
+{
+	char const* sourceCode = R"(
+		contract ClientReceipt {
+			event Deposit;
+			event Deposit(address _addr);
+			event Deposit(address _addr, uint _amount);
+			function deposit() {
+				Deposit();
+			}
+			function deposit(address _addr) {
+				Deposit(_addr);
+			}
+			function deposit(address _addr, uint _amount) {
+				Deposit(_addr, _amount);
+			}
+		}
+	)";
+	compileAndRun(sourceCode);
+	callContractFunction("deposit()");
+	BOOST_REQUIRE_EQUAL(m_logs.size(), 1);
+	BOOST_CHECK_EQUAL(m_logs[0].address, m_contractAddress);
+	BOOST_CHECK(m_logs[0].data.empty());
+	BOOST_REQUIRE_EQUAL(m_logs[0].topics.size(), 1);
+	BOOST_CHECK_EQUAL(m_logs[0].topics[0], dev::keccak256(string("Deposit()")));
+
+	callContractFunction("deposit(0xabcdeabcdeabcdeabcde)");
```

Comment:
You need to provide the signature in the string and the arguments as C++ arguments:
`callContractFunction("deposit(address)", u256("0xabcdeabcdeabcdeabcde"))`

And please always check the return value of the function. Furthermore, it might make sense to just return a fixed number inside the function, otherwise we cannot distinguish between an exception and just returning the empty string.